mhkit/wave/io/ndbc.py

```diff
@@ -0,0 +1,618 @@
+    if header[0].startswith("#"):
+        header[0] = header[0][1:]
+        header_commented = True
+    else:
+        header_commented = False
+        
+    # If second line is commented, indicate that units exist
+    if units[0].startswith("#"):
+        units_exist = True
+    else:
+        units_exist = False
+    
+    # Check if the time stamp contains minutes, and create list of column names 
+    # to parse for date
+    if header[4] == 'mm':
+        parse_vals = header[0:5]
+        date_format = '%Y %m %d %H %M'
+        units = units[5:]   #remove date columns from units
+    else:
+        parse_vals = header[0:4]
+        date_format = '%Y %m %d %H'
+        units = units[4:]   #remove date columns from units
+    
+    # If first line is commented, manually feed in column names
+    if header_commented:
+        data = pd.read_csv(file_name, sep='\s+', header=None, names = header,
+                           comment = "#", parse_dates=[parse_vals]) 
+    # If first line is not commented, then the first row can be used as header                        
+    else:
+        data = pd.read_csv(file_name, sep='\s+', header=0,
+                           comment = "#", parse_dates=[parse_vals])
+                             
+    # Convert index to datetime
+    date_column = "_".join(parse_vals)
+    data['Time'] = pd.to_datetime(data[date_column], format=date_format)
+    data.index = data['Time'].values
+    # Remove date columns
+    del data[date_column]
+    del data['Time']
+    
+    # If there was a row of units, convert to dictionary
+    if units_exist:
+        metadata = {column:unit for column,unit in zip(data.columns,units)}
+    else:
+        metadata = None
+
+    # Convert columns to numeric data if possible, otherwise leave as string
+    for column in data:
+        data[column] = pd.to_numeric(data[column], errors='ignore')
+        
+    # Convert column names to float if possible (handles frequency headers)
+    # if there is non-numeric name, just leave all as strings.
+    try:
+        data.columns = [float(column) for column in data.columns]
+    except:
+        data.columns = data.columns
+    
+    # Replace indicated missing values with nan
+    data.replace(missing_values, np.nan, inplace=True)
+    
+    return data, metadata
+
+
+def available_data(parameter,
+                   buoy_number=None, 
+                   proxy=None):  
+    '''
+    For a given parameter this will return a DataFrame of years, 
+    station IDs and file names that contain that parameter data.              
+    
+    Parameters
+    ----------
+    parameter: string
+        'swden'	:	'Raw Spectral Wave Current Year Historical Data'
+        'stdmet':   'Standard Meteorological Current Year Historical Data'
+    buoy_number: string (optional)
+        Buoy Number.  5-character alpha-numeric station identifier        
+    proxy: string (optional)
+        proxy url
+        
```

Comment:
@ssolson I have one more minor request, can you specify the format for the proxy setting? For example
`'http://proxy_server.com:port_number'`

Reply:
> @ssolson I have one more minor request, can you specify the format for the proxy setting? For example
> `'http://proxy_server.com:port_number'`

Request 2.0 made proxies dictionaries. This should be fixed in 
2f0c97b. Let me know if you are still having issues.